lib/matplotlib/axes/_axes.py

```diff
@@ -1141,6 +1141,11 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
         if len(positions) == 0:
             return []
 
+        # remove 'singular' keys from **kwargs dict in order to prevent them
+        # being lazily passed as LineCollection props, where they override the
+        # effect of the 'plural' versions (e.g. 'color' overrides 'colors')
+        cbook.strip_problem_kwargs(kwargs, 'color', 'linewidth', 'linestyle')
```

Comment:
This just discards the 'bad' kwargs which we can't do without a deprecation cycle.  There are users out there that have been using `color` that this will break.  This needs to automate the precedence cascade.